src/landpatterns/VirtualLP.stanza

```diff
@@ -868,24 +868,42 @@ which was to try and find lines and intersections of those lines.
 This seemed like it might be a bit more robust but at the cost
 of being more complex code wise.
 
+@param vp VirtualLP Scene Graph - we will search for pads here.
+@param side Board side to collect pads from.
+@param layer-spec Specify which layer is used to collect the shapes.
+By default, this function uses the `SolderMask(side)` layer.
+To select the copper on this side - use `false`. See {@link bounds}
+
 @throws ValueError if it encounter a number of columns that it can't
 handle. Specifically, this function can handle [1, 2, 4] columns of pads.
 This corresponds to 2-pin, dual-row, and quad land patterns, respectively.
 <DOC>
-public defn pad-interior-bounds (vp:VirtualLP, side:Side) -> Box:
+public defn pad-interior-bounds (vp:VirtualLP, side:Side -- layer-spec:LayerSpecifier|False = SolderMask(side)) -> Box:
 
-  val get-bounds = bounds{_, layer-spec = SolderMask(side)}
+  val get-bounds = bounds{_, layer-spec = layer-spec}
   val cols = identify-pad-columns(vp)
   val num-cols = length(cols)
   if num-cols == 1:
     ; This is a 2-pin component - ie a SMT, Radial or Axial
     val row0 = get-bounds $ get-pads-by-row(vp, 0)
     val row1 = get-bounds $ get-pads-by-row(vp, 1)
 
-    Box(
-      Point(left(row1), up(row1))
-      Point(right(row0), down(row0))
-    )
+    try:
+      Box(
+        Point(left(row1), up(row1))
+        Point(right(row0), down(row0))
+      )
+    catch (e:NegativeBoxError):
```

Comment:
why isn't the Box guaranteed to be valid? does `get-bounds` not enforce any order?

Reply:
Because if you use the soldermask openings instead of the copper geometry - the soldermask openings might overlap. This isn't exactly what we want but we have to design with the specifications of the fab house. 

If the soldermask openings overlap, then this algorithm will result in creating a negative box. I don't think we want to force  create a box (ie the absolute value of the box). This is more pressing for the silkscreen case - where a negative box would indicate that we can't put silkscreen here anyway (no soldermask is present). 

For the copper case - this would technically just be an error - because it would mean that there are two pads with overlapping copper.  In that case - we would just fix the copper pads.